src/Functions/hasAnyAllTokens.cpp

```diff
@@ -70,16 +70,29 @@ void FunctionHasAnyAllTokens<HasTokensTraits>::setSearchTokens(const std::vector
 namespace
 {
 
+/// Function input accept string, fixed string, array of string or array of fixed strings.
+bool isStringOrFixedStringOrArrayOfStringOrFixedString(const IDataType & type)
+{
+    if (isStringOrFixedString(type))
+        return true;
+
+    if (const auto * array_type = checkAndGetDataType<DataTypeArray>(&type); array_type)
```

Comment:
A question here: Does this test also covers the Maps? They are mentioned in the documentation bellow, but this check only mentions array?